src/main/java/app/chatbot/Storage.java

```diff
@@ -0,0 +1,135 @@
+package app.chatbot;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import app.task.Task;
+import app.task.TaskList;
+import app.task.TaskTypes;
+
+/**
+ * Handles Storage of data. Converts Task data saved in text file into
+ * Tasks the user interacts with during runtime. Saves Tasks into txt file
+ * and overwrites changes.
+ */
+public class Storage {
+    public static final String SEPARATOR_REGEX = " \\| ";
+    public static final String SEPARATOR = " | ";
+    public static final String SUB_SEPARATOR = ":";
+    private Path path;
+    private final File file;
+
+    /**
+     * Creates a new Storage object. A file is created at the stated path to store
+     * data. If the path location does not exist, the necessary files and directories
+     * are created.
+     *
+     * @param path the relative path for which the text data file will be stored
+     * @throws IOException
+     */
+    Storage(Path path) throws IOException {
+        this.path = path;
+        this.file = new File(path.toString());
+
+        // creates storage directory or data if they don't exist
+        this.file.getParentFile().mkdirs();
+        if (!Files.exists(path)) {
+            this.file.createNewFile();
+        }
+    }
+
+    /**
+     * Loads a single line in storage into a given task.TaskList.
+     *
+     * It is assumed that a line in storage follows the format specified here:
+     * taskSymbol | isDone | desc | addtl-arg1:value | addtl-arg2:value ...
+     *
+     * For example, a project meeting task.Event from 1pm to 3pm marked done:
+     * E | 1 | project meeting | from:1pm | to:3pm
+     *
+     * @param tl empty TaskList, assuming this is executed upon startup of app.
+     * @return totalSuccess, the number of lines that have been successfully loaded into the tl.
+     */
+    public int loadIntoTaskList(TaskList tl) throws Exception {
+
+
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(this.file));
```

Comment:
The `BufferedReader br` is not closed and can cause a resource leak. Maybe you can add `br.close()` right before you `return totalSuccess`?